OpenRA.Game/MiniYaml.cs

```diff
@@ -158,13 +188,12 @@ static List<MiniYamlNode> FromLines(IEnumerable<string> lines, string filename,
 
 			var levels = new List<List<MiniYamlNode>>();
 			levels.Add(new List<MiniYamlNode>());
-
 			var lineNo = 0;
-			foreach (var ll in lines)
+			foreach (var line in lines)
```

Comment:
I feel like transliterating the existing logic to spans would be far more valuable for avoiding a bunch of string copies. Might even make it easier to read as you could take slices as soon as you are ready rather than having to keep track of start and end indexes in a bunch of places.

That might even be worth an isolated PR.